contracts/src/WarpProtocolRegistry.sol

```diff
@@ -0,0 +1,198 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import "@subnet-evm-contracts/interfaces/IWarpMessenger.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+/**
+ * @dev Registry entry that represents a mapping between protocolAddress and version.
+ */
+struct ProtocolRegistryEntry {
```

Comment:
I wonder if this struct may be more confusing than it's worth in terms of ABI encode an instance of the struct rather than just a `(uint256,address)` tuple. Or is that the same encoding given that there are no variable length elements of the struct?

Reply:
In this case it looks like `abi.encode()` will be the same with these field types, tested by changing unit tests to encode without struct and still decoding with struct. 

The struct also makes initialization simpler by passing an array of structs, instead of two arrays of version and addresses separately, with additional checks like lengths are same etc.